Use Safari view controller for opening more links

repomies opened the pull request:
Adds consistency to how links are opened by always using SFSafariViewController inside the app.

repomies committed:
Add LinkHandler utility for opening links in SFSafariViewController

repomies committed:
Use LinkHandler to open all links

repomies committed:
Present Safari view controller from the topmost presented view controller

ghost commented:
Thanks for taking the time to unify these but sadly I have to decline this PR as it is because the link handling is (mostly) implemented like this by choice.

App-specific links (like terms, privacy policy) and THL-related links should open in SFSafariViewController but Omaolo links should be separately opened in the user's browser of choice. I said "mostly" earlier because I believe this has not been consistently implemented at the moment. If you'd like, you can help us by changing the PR to do that.

repomies commented:
Thanks, I thought the actual logic could be something like this! I'll rework the PR a bit so Omaolo links are opened outside the app but others, like the ones in the info section of Settings, go to SFSafariViewController.

repomies committed:
Add inApp bool param to link handler function to determine whether to use in-app Safari or not

repomies committed:
Open Omaolo links outside of app, others in app

repomies committed:
Don't try to use Safari for system settings URL

repomies commented:
Also caught a silly mistake I made earlier — system settings URL shouldn't be opened in SFSafariViewController :)

Commit:
Some architectural adjustments

- Implement the link handling in the responsible ViewController logic instead
- Show an error if the external link cannot be opened (eg. Safari being disabled)

solita-pekko reviewed (changes requested):
While reviewing the changes I noticed that English version of HomeLogoLinkURL points to the Finnish page, it should be changed to https://thl.fi/en/web/thlfi-en. Other than that looks good.

Commit:
Fix the English THL homepage link URL

Commit:
Update CHANGELOG.md
